src/fsevents/fsevents_stubs.c

```diff
@@ -261,20 +261,22 @@ static const FSEventStreamEventFlags action_mask =
     kFSEventStreamEventFlagItemCreated | kFSEventStreamEventFlagItemRemoved |
     kFSEventStreamEventFlagItemRenamed | kFSEventStreamEventFlagItemModified;
 
+static int count_set_bits(uint32_t bits) {
+  // XXX slow
+  int count = 0;
+  while (bits) {
+    count += (bits & 1);
+    bits >>= 1;
+  }
+  return count;
+}
+
 CAMLprim value dune_fsevents_action(value v_flags) {
   CAMLparam1(v_flags);
   CAMLlocal1(v_action);
 
   uint32_t flags = Int32_val(v_flags) & action_mask;
-  // XXX slow
-  int count = 0;
-  while (flags) {
-    count += (flags & 1);
-    flags >>= 1;
-  }
-
-  flags = Int32_val(v_flags);
-  if (count >= 2 || flags & kFSEventStreamEventFlagItemRenamed) {
+  if (flags & kFSEventStreamEventFlagItemRenamed || count_set_bits(flags) >= 2) {
```

Comment:
```suggestion
  if ((flags & kFSEventStreamEventFlagItemRenamed) || (count_set_bits(flags) >= 2)) {
```

I always have a double take with operator precedence